pkg/sdk/metrics/metrics.go

```diff
@@ -0,0 +1,74 @@
+package metrics
+
+import (
+	"sync"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	eventsMetricName           = "operator_event_types_total"
+	reconcileResultsMetricName = "operator_reconcile_results_total"
+	// EventTypeLabel - metric label for event type
+	EventTypeLabel = "type"
+	// EventTypeAdd - addition event label
+	EventTypeAdd = "add"
+	// EventTypeDelete - deletion event label
+	EventTypeDelete = "delete"
+	// EventTypeUpdate - update event label
+	EventTypeUpdate = "update"
+	// ReconcileResultLabel - metric label for event result
+	ReconcileResultLabel = "result"
+	// ReconcileResultSuccess - successful event label
+	ReconcileResultSuccess = "success"
+	// ReconcileResultFailure - failed event label
+	ReconcileResultFailure = "failure"
+)
+
+var (
+	once sync.Once
+)
+
+// Collector - metric collector for all the metrics the sdk will watch
+type Collector struct {
+	EventType       *prom.CounterVec
+	ReconcileResult *prom.CounterVec
+}
+
+// New - create a new Collector
+func New() *Collector {
+	return &Collector{
+		EventType: prom.NewCounterVec(prom.CounterOpts{
+			Name: eventsMetricName,
+			Help: "events that the sdk has recieved, segmented by type(add or delete or update)",
+		}, []string{EventTypeLabel}),
+		ReconcileResult: prom.NewCounterVec(prom.CounterOpts{
+			Name: reconcileResultsMetricName,
+			Help: "reconcilation events that the sdk has processed segmented by result(success or failure)",
+		}, []string{ReconcileResultLabel}),
+	}
+}
+
+// RegisterCollector - add collector safely to prometheus
+func RegisterCollector(c *Collector) {
+	once.Do(func() {
+		err := prom.Register(c)
+		if err != nil {
+			logrus.Errorf("unable to register collector with prometheus: %v", err)
+		}
+	})
+}
+
+// Describe returns all the descriptions of the collector
+func (s *Collector) Describe(ch chan<- *prom.Desc) {
```

Comment:
This should be `func (c *Collector)`. The [receiver name](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names) should be an abbreviation of the type.
Same for Collect below.